### PR TITLE
Added source model generation for EAV attribute input

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
**Description** (*)

Added possibilities choose or generate source model according to eav attribute input.
Screenshot 2021-04-09 at 15 26 20

1. Generate product attribute data patch without source model:
 
https://user-images.githubusercontent.com/29776509/114182248-15aa2080-994b-11eb-8391-6f09b3a2b1a8.mp4

2. Choose source model for some inputs:

https://user-images.githubusercontent.com/29776509/114182381-383c3980-994b-11eb-955a-bb2f6210db40.mp4

3. Generate source model:

https://user-images.githubusercontent.com/29776509/114182455-4d18cd00-994b-11eb-81c2-283f6a071cac.mp4

https://user-images.githubusercontent.com/29776509/114182480-52761780-994b-11eb-89eb-1c526db19033.mp4

**Fixed Issues (if relevant)**

1. Fixes magento/magento2-phpstorm-plugin#532

**Questions or comments**

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/magento/packages/eav/AttributeInput.java [hunk 1]
```diff
@@ -5,24 +5,24 @@
 
 package com.magento.idea.magento2plugin.magento.packages.eav;
 
-public enum AttributeInputs {
+public enum AttributeInput {
+    TEXT("text"),
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/magento/packages/eav/AttributeInput.java:9]: 👍 nice sorting

File: src/com/magento/idea/magento2plugin/magento/packages/uicomponent/AvailableSourcesByInput.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package com.magento.idea.magento2plugin.magento.packages.uicomponent;
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/magento/packages/uicomponent/AvailableSourcesByInput.java:1]: add copyright here please

File: src/com/magento/idea/magento2plugin/actions/generation/data/SourceModelData.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
+package com.magento.idea.magento2plugin.actions.generation.data;
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/data/SourceModelData.java:1]: Copyright

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/event/EavAttributeInputItemListener.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package com.magento.idea.magento2plugin.actions.generation.dialog.event;
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/event/EavAttributeInputItemListener.java:1]: Copyright

File: src/com/magento/idea/magento2plugin/actions/generation/generator/SourceModelGenerator.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
+package com.magento.idea.magento2plugin.actions.generation.generator;
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/SourceModelGenerator.java:1]: Add copyright here pls

File: src/com/magento/idea/magento2plugin/magento/files/SourceModelPhp.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
+package com.magento.idea.magento2plugin.magento.files;
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/magento/files/SourceModelPhp.java:1]: Add copyright here pls

File: tests/com/magento/idea/magento2plugin/actions/generation/generator/SourceModelGeneratorTest.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
+package com.magento.idea.magento2plugin.actions.generation.generator;
```
Comment (VitaliyBoyko) [inline tests/com/magento/idea/magento2plugin/actions/generation/generator/SourceModelGeneratorTest.java:1]: Add copyright here pls

**Review by VitaliyBoyko — CHANGES_REQUESTED** [review]

**Comment by ProkopovVitaliy** [comment]
@VitaliyBoyko copyright added for all mentioned files.

Thanks!

**Review by VitaliyBoyko — APPROVED** [review]
Great job. Thank you @ProkopovVitaliy

**Pull request merged**